src/plugins/tables/tables.js

```diff
@@ -418,10 +418,8 @@ $document.on( "click", ".wb-tables-filter [type='reset']", function( event ) {
 	$datatable.search( "" ).columns().search( "" ).draw();
 
 	$form.find( "select" ).prop( "selectedIndex", 0 );
-	$form.find( "input:checkbox" ).prop( "checked", false );
-	$form.find( "input:radio" ).prop( "checked", false );
-	$form.find( "input[type=date]" ).val( "" );
-
+	$form.find( "input:checkbox, input:radio" ).prop( "checked", false );
+	$form.find( ":input" ).not( ":button, :submit, :reset, :hidden, :checkbox, :radio" ).val( "" );
```

Comment:
@polmih can you validate if this behaviour match HTML5 spec.

Here the reference:
* https://html.spec.whatwg.org/multipage/form-elements.html#attr-button-type-reset 
* https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#resetting-a-form